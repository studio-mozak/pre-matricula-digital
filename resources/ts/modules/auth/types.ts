export interface Auth {
  level: number;
  name: string;
  schools: string[];
}
