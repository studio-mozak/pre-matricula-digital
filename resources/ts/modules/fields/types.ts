import { Nullable } from '@/types';

export type SetValues = (fields: Record<string, unknown>) => void;

export interface FieldPageOption {
  id?: string;
  name: Nullable<string>;
  weight: number;
}

export interface FieldPage {
  id: Nullable<string>;
  name: Nullable<string>;
  description: Nullable<string>;
  group: Nullable<'RESPONSIBLE' | 'STUDENT'>;
  type: Nullable<string>;
  internal: boolean;
  options: FieldPageOption[];
  deleteOptions?: string[];
  required?: boolean;
}

export interface ErrorResponse {
  message: string;
}
