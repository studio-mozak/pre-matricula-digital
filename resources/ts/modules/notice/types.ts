import { Nullable } from '@/types';

export interface Notice {
  id: Nullable<number>;
  text: Nullable<string>;
}

export interface NoticeSubmitResponse {
  errors: boolean;
  id?: number;
}
