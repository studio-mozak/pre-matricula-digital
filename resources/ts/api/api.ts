import axios, { AxiosInstance } from 'axios';

const token = window.config ? window.config.token : '';

export const http: AxiosInstance = axios.create({
  baseURL: String(import.meta.env.VITE_APP_API_URL || ''),
  headers: {
    Authorization: `Token ${token}`,
  },
});

axios.interceptors.response.use(
  (res) => {
    /**
     * If the response from the server has the "errors" object or the object "data" is empty,
     * here it is forced to go to the catch block automatically.
     */
    if (
      (res.data.errors || Object.keys(res.data).length === 0) &&
      !(res.data instanceof Blob)
    ) {
      return Promise.reject({
        response: {
          data: res.data,
        },
      });
    }

    return res;
  },
  (err) => Promise.reject(err)
);

export const rest = <T>(
  type: 'get' | 'post' | 'put' | 'delete',
  url: string,
  params?: object
): Promise<T> => http[type](url, params);

export const graphql = <T>(data: object, config?: object): Promise<T> =>
  axios.post('/graphql', data, {
    headers: {
      Authorization: `Bearer ${token}`,
    },
    ...config,
  });
