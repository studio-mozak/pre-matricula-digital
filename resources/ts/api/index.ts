export * from './api';
export * as City from './services/graphql/city';
export * as Resource from './services/graphql/resource';
