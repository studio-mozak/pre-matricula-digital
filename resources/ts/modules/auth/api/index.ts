/**
 * Exporting the Axios Instance
 */
export { http, graphql } from '@/api';

/**
 * Exporting all modules
 */
export * as AuthRest from '@/modules/auth/api/services/rest/auth';
