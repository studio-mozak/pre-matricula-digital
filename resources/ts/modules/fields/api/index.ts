/**
 * Exporting the Axios Instance
 */
export { http, graphql } from '@/api';

/**
 * Exporting all modules
 */
export * as Field from './services/graphql/field';
