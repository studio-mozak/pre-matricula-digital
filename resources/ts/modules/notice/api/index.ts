/**
 * Exporting the Axios Instance
 */
export { http, graphql } from '@/api';

/**
 * Exporting all modules
 */
export * as Notice from './services/graphql/notice';
