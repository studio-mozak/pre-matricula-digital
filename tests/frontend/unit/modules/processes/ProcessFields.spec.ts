import { VueWrapper, flushPromises, mount } from '@vue/test-utils';
import { ProcessField as ProcessFieldApi } from '@/modules/processes/api';
import { ProcessFieldList } from '@/modules/processes/types';
import ProcessFields from '@/modules/processes/components/ProcessFields.vue';
import VTooltip from 'v-tooltip';
import { expect, vi } from 'vitest';
import { AnalyticsInstance } from 'analytics';
import { analytics } from '@/packages';

interface ProcessFieldsProps {
  id?: string;
  newProcess: string;
}

vi.mock('@/modules/processes/api');
vi.mock('vue-router/dist/vue-router.mjs', () => ({
  useRouter: () => ({
    push: spyPush,
  }),
}));
vi.mock('@/packages');

const spyPush = vi.fn();

const processFieldMock: ProcessFieldList = {
  process: {
    fields: [
      { field: { id: '1' }, required: true, weight: 0, order: 1 },
      { field: { id: '3' }, required: true, weight: 0, order: 2 },
      { field: { id: '4' }, required: false, weight: 0, order: 3 },
      { field: { id: '2' }, required: true, weight: 0, order: 4 },
      { field: { id: '6' }, required: false, weight: 0, order: 5 },
    ],
    id: '1',
    name: 'Ed. Infantil (Nível III, Pré-Escola) Ens. Fundamental (1° ao 9° ano)',
  },
  responsibleFields: [
    {
      checked: true,
      group: 'RESPONSIBLE',
      id: '1',
      internal: 'responsible_name',
      mandatory: true,
      name: 'Nome do(a) responsável',
      open: false,
      options: [],
      order: 1,
      priority: false,
      required: true,
      type: 'TEXT',
      weight: 0,
    },
  ],
  studentFields: [
    {
      checked: true,
      group: 'STUDENT',
      id: '9',
      internal: 'student_name',
      mandatory: true,
      name: 'Nome',
      open: false,
      options: [],
      order: 1,
      priority: false,
      required: true,
      type: 'TEXT',
      weight: 0,
    },
  ],
};

const onMounted = async (props: ProcessFieldsProps) => {
  vi.mocked(ProcessFieldApi.list).mockResolvedValueOnce(processFieldMock);

  vi.mocked(analytics).mockImplementation(
    () =>
      ({
        page: vi.fn(),
      } as unknown as AnalyticsInstance)
  );

  const wrapper = mount(ProcessFields, {
    props,
    global: {
      stubs: [],
      directives: {
        tooltip: VTooltip,
      },
    },
  });

  await wrapper.vm.$nextTick();

  await flushPromises();

  return wrapper;
};

describe('ProcessFields', () => {
  let wrapper: VueWrapper<InstanceType<typeof ProcessFields>>;

  beforeEach(async () => {
    wrapper = await onMounted({
      newProcess: 'false',
      id: '1',
    });

    await wrapper.vm.$nextTick();
  });

  afterEach(() => {
    vi.clearAllMocks();
  });

  test('the component exists', () => {
    expect(wrapper).toBeTruthy();
  });

  test('should match snapshot', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  test('the initial variables must be loaded with the correct data', () => {
    expect(wrapper.vm.responsibleFields).toMatchObject(
      processFieldMock.responsibleFields
    );
    expect(wrapper.vm.studentFields).toMatchObject(
      processFieldMock.studentFields
    );
  });

  test('if I click on "go back" button, the user should be redirected to previous page', async () => {
    const button = wrapper.find('[data-test="button-back"]');

    await button.trigger('click');

    expect(spyPush).toBeCalled();
    expect(spyPush).toBeCalledTimes(1);
    expect(spyPush).toBeCalledWith({
      name: 'process.update',
      params: {
        id: '1',
        newProcess: 'false',
      },
    });
  });

  describe('submit', () => {
    test('should submit successfully', async () => {
      vi.mocked(ProcessFieldApi.post).mockResolvedValueOnce({});

      const button = wrapper.find('[data-test="button-proceed"]');

      await button.trigger('click');

      await wrapper.vm.$nextTick();

      await flushPromises();

      expect(spyPush).toHaveBeenCalledWith({
        name: 'process.periods',
        params: {
          id: '1',
          newProcess: 'false',
        },
      });
    });

    test('should not submit successfully', async () => {
      vi.mocked(ProcessFieldApi.post).mockRejectedValueOnce({});

      const button = wrapper.find('[data-test="button-proceed"]');

      await button.trigger('click');

      await wrapper.vm.$nextTick();

      await flushPromises();

      expect(spyPush).not.toBeCalled();
    });
  });
});
