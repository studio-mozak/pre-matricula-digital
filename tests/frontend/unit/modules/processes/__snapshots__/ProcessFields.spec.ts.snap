// Vitest Snapshot v1

exports[`ProcessFields > should match snapshot 1`] = `
<main
  class="container"
  style="max-width: 740px;"
>
  <h2
    class="font-muli-20-primary"
  >
     Escolha os campos solicitados no processo de pré-matrícula 
  </h2>
  <p>
     Sabendo que os campos obrigatórios do sistema já estão inclusos, selecione os campos criados que deseja inserir no formulário desse processo de pré-matrícula. 
  </p>
  <p>
    Você pode alterar a ordem dos campos, basta arrastá-los.
  </p>
  <!--v-if-->
  <p
    class="border-bottom mt-5 pb-3"
  >
     Dados do(a) responsável 
  </p>
  <div
    class="row"
  >
    <div
      class="col-12 col-md-6 can-drag mb-3"
      data-draggable="true"
    >
      <label
        class="toggle-checkbox"
      >
        <input
          checked=""
          disabled=""
          type="checkbox"
          value="1"
        />
        <div
          class="toggle-required toggle-text text-break overflow-hidden"
        >
          Nome do(a) responsável
        </div>
      </label>
      <!--v-if-->
    </div>
  </div>
  <p
    class="border-bottom mt-5 pb-3"
  >
     Dados do(a) aluno(a) 
  </p>
  <div
    class="row"
  >
    <div
      class="col-12 col-md-6 can-drag mb-3"
      data-draggable="true"
    >
      <label
        class="toggle-checkbox"
      >
        <input
          checked=""
          disabled=""
          type="checkbox"
          value="9"
        />
        <div
          class="toggle-required toggle-text text-break overflow-hidden"
        >
          Nome
        </div>
      </label>
      <!--v-if-->
    </div>
  </div>
  <div
    class="row mt-5"
  >
    <div
      class="offset-md-3 col-md-3"
    >
      <button
        class="x-btn x-btn-item non-selectable no-outline x-btn--outline x-btn--rectangle text-gray-500 border-gray-500 x-btn--actionable pmd-focusable pmd-hoverable x-btn--no-uppercase w-100"
        data-test="button-back"
        role="button"
        style="font-size: 20px;"
        tabindex="0"
        type="button"
      >
        <span
          class="x-btn__content text-center d-flex justify-content-center no-wrap text-no-wrap"
        >
          <span
            class="block"
          >
            Voltar
          </span>
        </span>
        <transition-stub
          appear="false"
          css="true"
          name="x-transition--fade"
          persisted="false"
        >
          <!---->
        </transition-stub>
      </button>
    </div>
    <div
      class="col-md-3 mt-3 mt-md-0"
    >
      <button
        class="x-btn x-btn-item non-selectable no-outline x-btn--standard x-btn--rectangle bg-primary text-white x-btn--actionable pmd-focusable pmd-hoverable x-btn--no-uppercase w-100 h-100 flex-row justify-content-center"
        data-test="button-proceed"
        role="button"
        style="font-size: 20px;"
        tabindex="0"
        type="button"
      >
        <span
          class="x-btn__content text-center d-flex justify-content-center no-wrap text-no-wrap"
        >
          <span
            class="block"
          >
            Prosseguir
          </span>
        </span>
        <transition-stub
          appear="false"
          css="true"
          name="x-transition--fade"
          persisted="false"
        >
          <!---->
        </transition-stub>
      </button>
    </div>
  </div>
</main>
`;
